R/print.R

```diff
@@ -65,20 +65,15 @@ print_install_summary <- function(x) {
     " {.timestamp {total_time}}"))
 }
 
-warn_for_loaded_packages <- function(pkgs, lib) {
-  if (length(maybe_bad <- intersect(pkgs, loadedNamespaces()))) {
-    loaded_from <- vcapply(
-      maybe_bad,
-      function(x) dirname(getNamespaceInfo(x, "path"))
+warn_for_loaded_packages <- function(pkgs, lib, loaded) {
+  if (is.null(loaded)) return()
+  bad <- intersect(pkgs, loaded)
+  bad <- setdiff(bad, "pak")
+  if (length(bad)) {
+    cli::cli_alert_warning(
+      "{.pkg {bad}} {?is/are} loaded in this session, \\
+       you probably need to restart R after the installation.",
```

Comment:
Would it be better to suggest restarting a fresh version of R _before_ the installation, rather than afterwards?

Reply:
I think this is fine for Unix, because if they load something in the profile, then they'll get the same message after a restart, and that is quite confusing.

On Windows we'll have a completely different message.